inlang/source-code/github-lint-action/src/main.ts

```diff
@@ -1,97 +1,388 @@
+		const prNumber = github.context.payload.pull_request?.number
 
-		// Debug logs are only output if the `ACTIONS_STEP_DEBUG` secret is true
-		core.debug(`I got all inputs: ${owner} ${repo} ${pr_number} ${token}`)
+		const repoBase = await openRepository(process.cwd(), {
+			nodeishFs: fs,
+			branch: github.context.payload.pull_request?.head.ref,
+		})
+		const projectListBase = await listProjects(repoBase.nodeishFs, process.cwd())
+		const results = projectListBase.map((project) => ({
+			projectPath: project.projectPath.replace(process.cwd(), ""),
+			errorsBase: [] as any[],
+			errorsHead: [] as any[],
+			installedRules: [] as InstalledMessageLintRule[],
+			reportsBase: [] as MessageLintReport[],
+			reportsHead: [] as MessageLintReport[],
+			lintSummary: [] as { id: string; name: string; count: number }[],
+			changedIds: [] as string[],
+			commentContent: "" as string,
+		}))
 
-		// @ts-ignore
-		const octokit = new github.getOctokit(token)
+		// Collect all reports from the base repository
+		for (const result of results) {
+			core.debug(`Checking project: ${result.projectPath}`)
+			const projectBase = await loadProject({
+				projectPath: process.cwd() + result.projectPath,
+				repo: repoBase,
+				appId: "app.inlang.ninjaI18nAction",
+			})
+			if (projectBase.errors().length > 0) {
+				if (result) result.errorsBase = projectBase.errors()
+				console.debug("Skip project ", result.projectPath, " in base repo because of errors")
+				continue
+			}
+			result.installedRules.push(...projectBase.installed.messageLintRules())
+			result.reportsBase.push(...projectBase.query.messageLintReports.getAll())
+		}
 
-		const { data: changedFiles } = await octokit.rest.pulls.listFiles({
-			owner,
-			repo,
-			pull_number: pr_number,
-		})
+		// Collect meta data for head and base repository
+		const headMeta = {
+			owner: github.context.payload.pull_request?.head.label.split(":")[0],
+			repo: github.context.payload.pull_request?.head.repo.name,
+			branch: github.context.payload.pull_request?.head.label.split(":")[1],
+			link: github.context.payload.pull_request?.head.repo.html_url,
+		}
+		const baseMeta = {
+			owner: github.context.payload.pull_request?.base.label.split(":")[0],
+			repo: repo,
+			branch: github.context.payload.pull_request?.base.label.split(":")[1],
+			link: github.context.payload.pull_request?.base.repo.html_url,
+		}
 
-		core.debug(`I got changed files: ${changedFiles}`)
+		const isFork = headMeta.owner !== baseMeta.owner
+		core.debug(`Is fork: ${isFork}`)
 
-		let diffData = {
-			additions: 0,
-			deletions: 0,
-			changes: 0,
+		// Prepare head repo
+		let repoHead
+		if (isFork) {
+			core.debug("Fork detected, cloning base repository")
+			process.chdir("../../../")
+			await cloneRepository(headMeta)
+			process.chdir(headMeta.repo)
+			repoHead = await openRepository(process.cwd(), {
+				nodeishFs: fs,
+			})
+		} else {
+			core.debug("Fork not detected, fetching and checking out base repository")
```

Comment:
not sure why this is needed aat all, the pr will run on the branch after the merge simulated allrady by default in github.